encoding/x/protobuf/protoc-gen-yarpc-go/main.go

```diff
@@ -0,0 +1,232 @@
+
+To use:
+	go get github.com/gogo/protobuf/protoc-gen-gogoslick
+	go get go.uber.org/yarpc/encoding/x/protobuf/protoc-gen-yarpc-go
+	protoc --gogoslick_out=. foo.proto
+	protoc --yarpc-go_out=. foo.proto
+*/
+package main
+
+// TODO: there is some crazy bug with protobuf that if you declare:
+//
+//   func bar() (kv.GetValueResponse, error) {
+//     return nil, errors.New("nil response and non-nil error")
+//   }
+//
+//   func foo() (proto.Message, error) {
+//     return bar()
+//   }
+//
+//   response, err := foo()
+//   fmt.Printf("%v %v\n", response, response == nil)
+//
+// This will print "<nil>, false". If you try to do something with response (ie call a function on it),
+// if will panic because response is nil. Something similar happens in golang/protobuf
+// too, so this is insane. If in bar(), you do response == nil, it will be true.
+// The generated code handles this.
+
+import (
+	"fmt"
+	"log"
+	"text/template"
+
+	"go.uber.org/yarpc/internal/protoplugin"
+)
+
+const tmpl = `{{$packagePath := .GoPackage.Path}}
+// Code generated by protoc-gen-yarpc-go
+// source: {{.GetName}}
+// DO NOT EDIT!
+
+package {{.GoPackage.Name}}
+
+import (
+	{{range $i := .Imports}}{{if $i.Standard}}{{$i | printf "%s\n"}}{{end}}{{end}}
+
+	{{range $i := .Imports}}{{if not $i.Standard}}{{$i | printf "%s\n"}}{{end}}{{end}}
+)
+
+{{range $service := .Services }}
+// {{$service.GetName}}Client is the client-side interface for the {{$service.GetName}} service.
+type {{$service.GetName}}Client interface {
+	{{range $method := unaryMethods $service}}{{$method.GetName}}(context.Context, *{{$method.RequestType.GoType $packagePath}}, ...yarpc.CallOption) (*{{$method.ResponseType.GoType $packagePath}}, error)
+	{{end}}
+	{{range $method := onewayMethods $service}}{{$method.GetName}}(context.Context, *{{$method.RequestType.GoType $packagePath}}, ...yarpc.CallOption) (yarpc.Ack, error)
+	{{end}}
+}
+
+// New{{$service.GetName}}Client builds a new client for the {{$service.GetName}} service.
+func New{{$service.GetName}}Client(clientConfig transport.ClientConfig) {{$service.GetName}}Client {
+	return &_{{$service.GetName}}Caller{protobuf.NewClient("{{$service.GetName}}", clientConfig)}
+}
+
+// {{$service.GetName}}Server is the server-side interface for the {{$service.GetName}} service.
+type {{$service.GetName}}Server interface {
+	{{range $method := unaryMethods $service}}{{$method.GetName}}(context.Context, *{{$method.RequestType.GoType $packagePath}}) (*{{$method.ResponseType.GoType $packagePath}}, error)
+	{{end}}
+	{{range $method := onewayMethods $service}}{{$method.GetName}}(context.Context, *{{$method.RequestType.GoType $packagePath}}) error
+	{{end}}
+}
+
+// Build{{$service.GetName}}Procedures prepares an implementation of the {{$service.GetName}} service for registration.
+func Build{{$service.GetName}}Procedures(server {{$service.GetName}}Server) []transport.Procedure {
```

Comment:
Instead of

    dispatcher.Register(kv.BuildKeyValueProcedures(keyValueHandler))
    
What about,

    dispatcher.Register(kv.NewKeyValueService(keyValueHandler))
    
I know it's not technically a new service or server, IMO it reads easier.